FWCore/ParameterSet/python/Types.py

```diff
@@ -118,14 +153,13 @@ def _dumpPythonName():
         return 'obsolete'
 
 class _AllowedParameterTypes(object):
-    def __init__(self, *args):
+    def __init__(self, *args, **kargs):
         self.__dict__['_AllowedParameterTypes__types'] = args
-        self.__dict__['_AllowedParameterTypes__value'] = None
+        self.__dict__['_default'] = None
         self.__dict__['__name__'] = self.dumpPython()
+        if len(kargs):
+            self.__dict__['_default'] = self._setValueWithType(kargs['default'])
```

Comment:
IIUC, if the caller uses an incorrect keyword argument, the error is about the `kargs` not having an element `'default'` rather than about an unexpected keyword argument. How about something along
```py
def __init__(self, *args, default=None):
    # <snip>
    if default is not None:
        self.__dict__['_default'] = self._setValueWithType(default)
```
?

Reply:
changed